oeps/oep-00XX-git-best-practices.rst

```diff
@@ -0,0 +1,107 @@
+
++-----------------+--------------------------------------------------------+
+| OEP             | :doc:`OEP-XXXX </oeps/oep-XXXX-bp-git-best-practixes`  |
++-----------------+--------------------------------------------------------+
+| Title           | Git best practices                                     |
++-----------------+--------------------------------------------------------+
+| Last Modified   | 2019-07-18                                             |
++-----------------+--------------------------------------------------------+
+| Authors         | Joseph Mulloy                                          |
+|                 |                                                        |
++-----------------+--------------------------------------------------------+
+| Arbiter         | <Arbiter's real name and email address>                |
++-----------------+--------------------------------------------------------+
+| Status          | Draft                                                  |
++-----------------+--------------------------------------------------------+
+| Type            | Best Practice                                          |
++-----------------+--------------------------------------------------------+
+| Created         | 2019-07-18                                             |
++-----------------+--------------------------------------------------------+
+| `Review Period` | <start - target end dates for review>                  |
++-----------------+--------------------------------------------------------+
+| `Resolution`    | <links to any discussions where the final              |
+|                 | status was decided>                                    |
++-----------------+--------------------------------------------------------+
+| `References`    | <links to any other relevant discussions               |
+|                 | or relevant related materials>                         |
++-----------------+--------------------------------------------------------+
+
+Abstract
+========
+
+This OEP outlines git best practices that if followed increase the usability
+of the git commit history.
+
+Motivation
+==========
+
+Git history is often used to understand the motivations and context in which a
+change was made. When commit messages are descriptive and link to a ticket it's
+possible to understand why a change was made. When the message is terse it's
+much more difficult to understand the change.
+
+Specification
+=============
+
+The specification describes the technical details of the Architecture, Best
+Practice or Process proposed by the OEP. If the proposal includes a new API,
+specify its syntax and semantics.
+
+Commit messages
+---------------
+* The first line should be no longer than 50 characters
+* The second line should be no longer than 72 characters
+* The first line should concisely describe the reason for the change
+* The first line should include the JIRA Issue ID, if there is a JIRA Issue
```

Comment:
For me, I think the most important part of the git message is explaining the "why", and to be able to do so without relying on the ticket for context. Including a reference to the ticket is great. But some tickets are not public for privacy/security reasons. Some tickets are not public because we still have trouble with all the permissions of JIRA. Some tickets don't work any more because they were part of our last JIRA system. Some tickets don't exist because they were from when we were using LightHouse. Who knows what we'll be forced to switch to in three years when Capital One buys Atlassian?

Making guidelines about past vs. present tense or character width guidelines are good things. But I think we should emphasize the content of the message itself and give some examples. I'd much rather be looking at a badly formatted but informative commit message than a one liner reference to a ticket.